src/harness/testcases/WINNF_FT_S_PCR_testcase.py

```diff
@@ -0,0 +1,354 @@
+import time
+import subprocess
+import sas_testcase
+from shapely import ops
+from full_activity_dump_helper import getFullActivityDumpSasUut
+from reference_models.ppa import ppa
+from reference_models.geo import drive, utils
+from util import configurable_testcase, loadConfig, \
+     makePalRecordsConsistent, writeConfig, getCertificateFingerprint
+TEST_CONFIG_COMMAND = 'cd reference_models;python test_config.py'
+SAS_TEST_HARNESS_URL = 'https://test.harness.url.not.used/v1.2'
+
+
+def isPpaWithinServiceArea(pal_records, ppa_zone_geometry):
+  """Check if the ppa zone geometry with in service area then return True.
+
+  Checks the ppa zone geometry's boundary and interior intersect only with the
+  interior of the service area (not its boundary or exterior).
+
+  Args:
+    pal_records: A list of pal records to compute service area based on
+      census_tracts.
+    ppa_zone_geometry: A PPA polygon dictionary in GeoJSON format.
+
+  Returns:
+    A value is the boolean with the value as True if the ppa zone geometry's
+      boundary and interior intersect with in the interior of the service
+      area otherwise value as false.
+
+  """
+  # Get the census tract for each pal record and convert it to Shapely
+  # geometry.
+  census_tracts_for_pal = [
+      utils.ToShapely(drive.census_tract_driver.GetCensusTract(
+          pal['license']['licenseAreaIdentifier'])
+                      ['features'][0]['geometry']) for pal in pal_records]
+  pal_service_area = ops.cascaded_union(census_tracts_for_pal)
+
+  # Convert GeoJSON dictionary to Shapely object.
+  ppa_zone_shapely_geometry = utils.ToShapely(ppa_zone_geometry)
+
+  return ppa_zone_shapely_geometry.buffer(-1e-6).within(pal_service_area)
+
+
+def assertRegConditionalsForPpaRefModel(registration_request,
+                                        conditional_registration_data):
+  """Check the REG Conditionals for PPA creation model and raises an exception.
+
+  Performs the assert to check installationParam present in
+  registrationRequests or conditional registration data and raises an exception.
+  PpaCreationModel requires the input registrationRequests to have
+  'installationParam'. But this parameter is removed for devices where
+  conditionals are pre-loaded. Adding the 'installationParam' into
+  registrationRequests by taking the corresponding values from
+  conditionalRegistrationData.
+
+  Args:
+    registration_request: A list of individual CBSD registration
+      requests (each of which is itself a dictionary).
+    conditional_registration_data: A list of individual CBSD registration
+      data that need to be preloaded into SAS (each of which is a dictionary).
+      data that need to be preloaded into SAS (each of which is a dictionary).
+      the fccId and cbsdSerialNumber fields are required, other fields are
+      optional but required for ppa reference model.
+
+  Raises:
+    It will throws an exception if the installationParam object and required
```

Comment:
Specify the exception type. In this case it is a standard one: 
```
Raises:
   Exception: if the installationParam object ...
```

Reply:
Specify   the Exception type in doc string and update accordingly.